File: 1-Two-Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        
        for (int i = 0; i < nums.size(); i++) {
            int otherNum = target - nums[i];
            if (map.count(otherNum)) {
                return { map[otherNum], i };
            }
            
            map[nums[i]] = i;
        }

        return {};
    }
};

File: 1-Two-Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict: [Int : Int] = [:]
        for (i, num) in nums.enumerated() {
            if let otherIndex = dict[target - num] {
                return [i, otherIndex]
            }
            dict[num] = i
        }
        return []
    }
}